content/client-lib-development-guide/features.textile

```diff
@@ -1697,13 +1735,40 @@ class RealtimeChannel:
   unsubscribe(String, (Message) ->) // RTL8a
   setOptions(options: ChannelOptions) => io // RTL16
 
+class BatchOperations:
+  publish([BatchSpec]) => BatchResult<BatchPublishResponse> // BO2a
+  publish(BatchSpec) => BatchResult<BatchPublishResponse> // BO2a
+  getPresence([String]) => BatchResult<BatchPresenceResponse> // BO2b
+
+class BatchResult<T>:
+  error: ErrorInfo? // BPA2b
+  responses: []T? // BPA2a
+
+class BatchPublishResponse:
+  channel: String // BPB2a
+  messageId: String? // BPB2b
+  error: ErrorInfo? // BPB2c
+
+class BatchPresenceResponse:
+  channel: String // BPD2a
+  presence: []BatchPresence // PBD2b
```

Comment:
I think this should be `[]PresenceMessage`, not sure an extra type is needed.

Reply:
PresenceMessage has far more fields on it than what appears in the response for a batch presence request. If those fields are all optional or can be retrieved from context then I would be happy to use that type instead.

Reply:
> PresenceMessage has far more fields on it than what appears in the response for a batch presence request

Can you give an example? I'm not sure why retrieving presence for multiple channels in one request should return less fields than retrieving for a single channel, if that is the case it sounds more like a problem with the API that we should fix rather than something we should be encoding into the type system.

Reply:
According to the spec, PresenceMessage has `connectionId`, `id` and `timestamp` along with `action` and `clientId` which are the only two fields that the REST API returns for batch presence.  
[Docs link for PresenceMessage](https://ably.com/docs/api/rest-api#presence) | [Docs link for BatchPresence](https://ably.com/docs/rest/batch#presence-success)

Reply:
> PresenceMessage has connectionId, id and timestamp along with action and clientId which are the only two fields that the REST API returns for batch presence.

The docs may seem to indicate that, but it's not accurate, you get the same fields from both endpoints:

```
$ curl -u "${ABLY_API_KEY}" "https://rest.ably.io/channels/test1/presence"
[
        {
                "id": "LnUWZJf1G-:0:0",
                "clientId": "lmars",
                "connectionId": "LnUWZJf1G-",
                "timestamp": 1653686295156,
                "data": "",
                "action": 1
        }
]

$ curl -u "${ABLY_API_KEY}" "https://rest.ably.io/presence?channel=test1"
[
        {
                "channel": "test1",
                "presence": [
                        {
                                "id": "LnUWZJf1G-:0:0",
                                "clientId": "lmars",
                                "connectionId": "LnUWZJf1G-",
                                "timestamp": 1653686295156,
                                "data": "",
                                "action": 1
                        }
                ]
        }
]
```

Reply:
I've updated to use `PresenceMessage` in [384b4fe](https://github.com/ably/docs/pull/1450/commits/384b4fe48a169b3f7b40b839ee25611035a7965b). I've also created https://github.com/ably/docs/issues/1451 to address this on the docs side.